packages/ember-application/tests/system/visit_test.js

```diff
@@ -35,6 +37,66 @@ moduleFor('Application - visit()', class extends ApplicationTestCase {
     );
   }
 
+  [`@test does not add serialize-mode markers by default`](assert) {
+    let templateContent = '<div class="foo">Hi, Mom!</div>';
+    this.addTemplate('index', templateContent);
+    let rootElement = document.createElement('div');
+
+    let bootOptions = {
+      isBrowser: false,
+      rootElement
+    };
+
+    ENV._APPLICATION_TEMPLATE_WRAPPER = false;
+    return this.visit('/', bootOptions).then(()=> {
+      assert.equal(rootElement.innerHTML, templateContent, 'without serialize flag renders as expected');
+    });
+  }
+
+  [`@test _renderMode: rehydration`](assert) {
+    assert.expect(2);
+
+    let indexTemplate = '<div class="foo">Hi, Mom!</div>';
+    this.addTemplate('index', indexTemplate);
+    let rootElement = document.createElement('div');
+
+    let bootOptions = {
+      isBrowser: false,
+      rootElement,
+      _renderMode: 'serialize'
+    };
+
+    ENV._APPLICATION_TEMPLATE_WRAPPER = false;
+
+    return this.visit('/', bootOptions)
+      .then((instance) => {
+        assert.equal(
+          instance.rootElement.firstChild.nodeValue,
+          '%+b:0%',
+          'glimmer-vm comment node was not found'
+        );
+      }).then(() =>{
+        return this.runTask(()=>{
+          this.applicationInstance.destroy();
+          this.applicationInstance = null;
+        });
+      }).then(() => {
+        bootOptions = {
+          isBrowser: false,
+          rootElement,
+          _renderMode: 'rehydrate'
+        };
+
+        this.application.visit('/', bootOptions).then(instance => {
```

Comment:
@rwjblue turns out this is actually necessary.  When you remove the `this.application` and just call `visit` directly we get an error saying that you tried to use an object that is destroyed (sorry I don't have the exact wording of the error) . Either way nbd.

Reply:
Seems fine, but I still don't like it 😝